### DISCUSSION TITLE
Add the option to take SEO metadata for redirections

### DISCUSSION THREAD

**Discussion by nicolas-graves** [wagtail/wagtail#11391] [Ideas]
### Is your proposal related to a problem?

 My organization often needs to redirect to a pdf file with SEO metadata (so that social media gets an image and a description). The current redirection rules are not sufficient to do this kind of redirection. 
We use this type of page : 

```python
class RedirectionPage(Page):
    class Meta:
        verbose_name = "Page de redirection"

    lien_redirection = RichTextField(null=True)

    content_panels = Page.content_panels + [
        FieldPanel("lien_redirection"),
    ]

    feed_image = models.ForeignKey(
        "wagtailimages.Image",
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name="+",
        verbose_name="Vignette pour les réseaux sociaux",
    )

    promote_panels = Page.promote_panels + [FieldPanel("feed_image")]
```

and maybe that's the right way to do this. 

### Describe the solution you'd like

However, if this could be included in a way as an option for the redirections feature, it would perfectly match our use case. This would enable us to have a social media post with metadata, redirecting to a Document. 

### Describe alternatives you've considered

The RedirectionPage.

**Comment by thibaudcolas** [comment]
Hi @nicolas-graves, thanks for raising this. I see there hasn’t been much feedback on your idea so I thought I’d convert to Discussions to make it more visible.

I think this is pretty cool as a site-specific customization or package, probably not as built-in feature. As far as built-in features, perhaps we could support more extensibility of redirects, but not _exactly_ the requirements listed here.
